### PR TITLE
Hl 963 ps

### PR DESCRIPTION
**Pull request by rikuke**
## Description :sparkles:
Submitting an application manually as a handler does not create rows into the` bf_calculation_paysubsidy `table in the database, because the JSON payload it submits has an empty `pay_subsidies` list evaluates to not` None `which causes a branch in the calculation logic to skip the database insert.  This PR  fixes the issue by checking if the application is submitted by a handler and then makes sure that `pay_subsidy_data `is `None `and `pay_subsidies` is is not present in the `validated_data` dict.
## Testing :alembic:
Pytest: pytest applications/tests/test_handler_application.py
Manually: create a new application as  a handler and then as a handler process the application.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2281.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2281.dev.hel.ninja :laughing::tada::tada::tada:

File: backend/benefit/applications/api/v1/serializers/application.py [hunk 1]
```diff
@@ -1133,6 +1133,7 @@ def _validate_attachments(self, instance):
                     required_attachment_types.remove(attachment.attachment_type)
 
         if required_attachment_types:
+            print(required_attachment_types)
```
Comment (sirtawast) [inline backend/benefit/applications/api/v1/serializers/application.py:1136]: Remove print. Otherwise LGTM.

**Review by sirtawast — CHANGES_REQUESTED** [review]

File: backend/benefit/applications/api/v1/serializers/application.py (hunk 1 above)
Comment (rikuke) [reply backend/benefit/applications/api/v1/serializers/application.py:1136]: whoops

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2281.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2281.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2281.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2281.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://helsinkilisa-pr2281.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://helsinkilisa-ui-handler-pr2281.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**APPLICANT branch is deployed to platta: https://helsinkilisa-ui-pr2281.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://helsinkilisa-ui-pr2281.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**